drift_dev/lib/src/cli/commands/make_migrations.dart

```diff
@@ -0,0 +1,505 @@
+    }
+
+    final stepByStepTests = migrations
+        .map((e) => e.testStepByStepMigrationCode(dbName, dbClassName));
+
+    final code = """
+// ignore_for_file: unused_local_variable, unused_import
+// GENERATED CODE, DO NOT EDIT BY HAND.
+import 'package:drift/drift.dart';
+import 'package:drift_dev/api/migrations.dart';
+import 'package:$packageName/$relativeDbPath';
+import 'package:test/test.dart';
+import 'schemas/schema.dart';
+
+${stepByStepTests.map((e) => e.imports).expand((imports) => imports).toSet().join('\n')}
+
+void main() {
+  driftRuntimeOptions.dontWarnAboutMultipleDatabases = true;
+  late SchemaVerifier verifier;
+
+  setUpAll(() {
+    verifier = SchemaVerifier(GeneratedHelper());
+  });
+
+  ${stepByStepTests.map((e) => e.test).join('\n')}
+
+}
+""";
+    final testFile = File(p.join(testDir.path, 'migration_test.dart'));
+    if (testFile.existsSync()) {
+      cli.logger.fine(
+          '$dbName: Updated test in ${blue.wrap(p.relative(testFile.path))}');
+    } else {
+      cli.logger.info(
+          '$dbName: Generated test in ${blue.wrap(p.relative(testFile.path))}.\n'
+          'Run this test to validate that your migrations are written correctly. ${yellow.wrap("dart test ${blue.wrap(p.relative(testFile.path))}")}');
+    }
+
+    writeTasks[testFile] = code;
+  }
+
+  /// The json file where the schema for the current version of the database is stored
+  File driftSchemaFile(int version) {
+    return File(p.join(schemaDir.path, 'drift_schema_v$version.json'));
+  }
+
+  File testUtilityFile(int version) {
+    return File(p.join(testDatabasesDir.path, 'schema_v$version.dart'));
+  }
+
+  /// Generated file where the step by step migration code is stored
+  File get stepsFile {
+    return File(dbClassFile.absolute.path
+        .replaceFirst(RegExp(r'\.dart$'), '.steps.dart'));
+  }
+}
+
+/// A writer that generates the code for a migration from one schema version to another
+class _MigrationWriter {
+  final List<DriftTable> tables;
+  final int from;
+  final int to;
+
+  _MigrationWriter(this.tables, {required this.from, required this.to});
+
+  /// Create  list of migration writers from a map of schema versions
+  /// A migration writer is created for each pair of schema versions
+  /// e.g (1,2), (2,3), (3,4) etc
+  static List<_MigrationWriter> fromSchema(Map<int, ExportedSchema> schemas) {
+    final result = <_MigrationWriter>[];
+    if (schemas.length < 2) {
+      return result;
+    }
+    final versions = schemas.keys.toList()..sort();
+    for (var i = 0; i < versions.length - 1; i++) {
+      final (from, fromSchema) = (versions[i], schemas[versions[i]]!);
+      final (to, toSchema) = (versions[i + 1], schemas[versions[i + 1]]!);
+      final fromTables = fromSchema.schema.whereType<DriftTable>();
+      final toTables = toSchema.schema.whereType<DriftTable>();
+      final commonTables = fromTables.where(
+          (table) => toTables.any((t) => t.schemaName == table.schemaName));
```

Comment:
Why are we restricting schema elements to tables here?

Reply:
What else is insertable?
I'm even realizing that custom row classes may be an issue if they don't implement `Insertable`, how do you suppose we deal with that?

Reply:
never mind, seems it works fine with custom row lasses, but what else could be tested?

Reply:
_ping_ 😏

Reply:
Ah sorry, I thought this was responsible for also generating the schema code somehow.

The name and dartdocs of this class are a bit confusing since it essentially emits a test file for a migration? Maybe we should call it `_MigrationTestEmitter` or something, and mention that it's responsible for emitting a default test file verifying data integrity.

Reply:
Great point, renamed